On illumos, mbrtoc32() fails for every character in the zh_CN.GB18030 locale
because there is no GB18030 to UTF-32LE iconv module (illumos 18286), so
setlocale() fails when its result is decoded. Use ru_RU.KOI8-R instead.

diff -wpruN --no-dereference '--exclude=*.orig' a~/Lib/test/test_socket.py a/Lib/test/test_socket.py
--- a~/Lib/test/test_socket.py	1970-01-01 00:00:00
+++ a/Lib/test/test_socket.py	1970-01-01 00:00:00
@@ -1736,7 +1736,7 @@ class GeneralModuleTests(unittest.TestCa
                          'needs socket.AI_NUMERICSERV')
     @support.run_with_locales('LC_ALL',
         'uk_UA.KOI8-U', 'uk_UA', 'ja_JP.eucJP', 'ja_JP.SJIS', 'ja_JP',
-        'ko_KR.eucKR', 'zh_CN.GB18030', 'el_GR.ISO8859-7',
+        'ko_KR.eucKR', 'ru_RU.KOI8-R', 'el_GR.ISO8859-7',
         'de_DE.ISO8859-1', 'ja_JP.UTF-8',
         '')
     def test_getaddrinfo_localized_error(self):
